src/features/posts.ts

```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const getPosts = createAsyncThunk<Post[], User['id']>(
+  'posts/fetch',
+  userId => getUserPosts(userId),
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(getPosts.fulfilled, (state, action) => {
+        return {
+          ...state,
+          posts: action.payload,
+        };
```

Comment:
In your fulfilled case, you're replacing the entire posts array with the new data fetched from the API. This might not be the intended behavior if you want to append new posts to the existing ones. If you want to append new posts, you should use the spread operator on the existing state.posts and the action.payload.